tests/client/disco/components/TestAddon.js

```diff
@@ -220,16 +227,40 @@ describe('<Addon />', () => {
         payload: {guid},
       }));
     });
+
+    it('sets status to error on onDownloadFailed', () => {
+      const dispatch = sinon.spy();
+      const guid = '{my-addon}';
+      const i18n = getFakeI18nInst();
+      const handler = makeProgressHandler(dispatch, guid, i18n);
+      handler({state: 'STATE_SOMETHING'}, {type: 'onDownloadFailed'});
+      assert(dispatch.calledWith({
+        type: 'INSTALL_ERROR',
+        payload: {guid, errorMessage: 'Download failed. Please check your connection.'},
```

Comment:
I suppose an alternative to passing `errorMessage` around would be to pass a constant and key a translation off the constant closer to the error message display. That would avoid all the passing i18n around.